src/components/Header/Header.tsx

```diff
@@ -1,30 +1,46 @@
-import { Link, useLocation } from 'react-router-dom';
+'use client';
+
 import './Header.css';
+import { Link, usePathname, useRouter } from '@/i18n/navigation';
 import { ThemeSwitcher } from '../ThemeSwitcher';
+import { useLocale, useTranslations } from 'next-intl';
+
+export const Header = () => {
+  const pathname = usePathname();
+  const locale = useLocale();
+  const router = useRouter();
+  const t = useTranslations('Header');
 
-const Header = () => {
-  const location = useLocation();
+  const switchLanguage = () => {
+    const newLocale = locale === 'en' ? 'ru' : 'en';
+    router.replace(pathname, { locale: newLocale });
+  };
```

Comment:
In such realisation you clean query params if there are exist in the url after switching the language
You need to prepare the url before replacing the locale
```
    const paramsString = params.toString();
    const newUrl = paramsString ? `${pathname}?${paramsString}` : pathname;
    router.replace(newUrl, { locale: newLocale });
```
where 
```
  import { useSearchParams } from 'next/navigation';

  const searchParams = useSearchParams();
  const params = new URLSearchParams(searchParams);
```